erpnext/manufacturing/doctype/work_order/work_order.py

```diff
@@ -706,3 +706,13 @@ def get_work_order_operation_data(work_order, operation, workstation):
 	for d in work_order.operations:
 		if d.operation == operation and d.workstation == workstation:
 			return d
+
+@frappe.whitelist()
+def get_items_with_bom(doctype, txt, searchfield, start, page_len, filters):
+	items = frappe.get_all("Item", filters=[('default_bom', '!=', 'NULL')], fields=["name", "default_bom"])
```

Comment:
Why don't you get the distinct item directly from BOM instead of item? Also put the limit_start=start, limit_page_length=page_length in the get_all

Reply:
done at client side